integrationtest/showcase/library/src/main/java/org/hibernate/search/integrationtest/showcase/library/repository/impl/DocumentRepositoryImpl.java

```diff
@@ -144,4 +147,25 @@ public long count() {
 
 		return query.getResultList();
 	}
+
+	@Override
+	public List<String> getAuthorsOfBooksHavingTerms(String terms, SortOrder order) {
+		FullTextSearchTarget<Document> target = entityManager.search( Document.class );
+		FullTextQuery<List<?>> query = target.query()
+				.asProjections(
```

Comment:
Use `asProjection` (without the trailing "s") and you'll get a `FullTextQuery<String>` and will not need the awkward stream/map/collect below.

Also, you could use lambdas here, too:

```
.asProjections( f -> f.field( "author", String.class ) )
```

Reply:
Good api ;)